src/main/java/formflow/library/interceptors/SessionContinuityInterceptor.java

```diff
@@ -0,0 +1,117 @@
+package formflow.library.interceptors;
+
+import formflow.library.config.FlowConfiguration;
+import formflow.library.exceptions.LandmarkNotSetException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.core.Ordered;
+import org.springframework.stereotype.Component;
+import org.springframework.util.AntPathMatcher;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+/**
+ * This interceptor prevents users from jumping to random pages in a flow.
+ */
+@Component
+@Slf4j
+@ConditionalOnProperty(name = "form-flow.session-continuity-interceptor.enabled", havingValue = "true")
+public class SessionContinuityInterceptor implements HandlerInterceptor, Ordered {
+
+  public static final String FLOW_PATH_FORMAT = "/flow/{flow}/{screen}";
+  public static final String NAVIGATION_FLOW_PATH_FORMAT = "/flow/{flow}/{screen}/navigation";
+
+  List<FlowConfiguration> flowConfigurations;
+
+  public SessionContinuityInterceptor(List<FlowConfiguration> flowConfigurations) {
+    this.flowConfigurations = flowConfigurations;
+  }
+
+  /**
+   * @param request  current HTTP request
+   * @param response current HTTP response
+   * @param handler  chosen handler to execute, for type and/or instance evaluation
+   * @return Boolean True - allows the request to proceed to the ScreenController, False - stops the request from reaching the
+   * Screen Controller.
+   * @throws Exception
+   */
+  @Override
+  public boolean preHandle(HttpServletRequest request, @NotNull HttpServletResponse response, @NotNull Object handler)
+      throws Exception {
+    try {
+      String pathFormat = request.getRequestURI().contains("navigation") ? NAVIGATION_FLOW_PATH_FORMAT : FLOW_PATH_FORMAT;
+      Map<String, String> parsedUrl = new AntPathMatcher().extractUriTemplateVariables(pathFormat, request.getRequestURI());
+      String redirect_url = "/";
+      String screen = parsedUrl.get("screen");
+      String flow = parsedUrl.get("flow");
+      HttpSession session = request.getSession(false);
+      FlowConfiguration flowConfiguration = flowConfigurations.stream().filter(fc -> fc.getName().equals(flow)).findFirst()
+          .orElse(null);
```

Comment:
This block of variables is a bit dense. Some suggestions for what I am used to:
* `redirect_url` seems like a class level variable, or maybe something that should be configured as part of each landmark config
* `screen` and `session` could be moved further down in the code to right before they are used - clearer to read and ensures they are only declared if needed
   * I might also inline `screen` since it is only used once
* `flow` is also only used once, but the line is rather long already. Here is the way I normally make `stream()` lines more readable:
```
FlowConfiguration flowConfiguration = flowConfigurations.stream()
          .filter(fc -> fc.getName().equals(parsedUrl.get("flow")))
          .findFirst()
          .orElse(null);
``` 
basically every chained dot method on the stream gets its own line

Reply:
I like to write streams this way as well, as I find it more readable.   I'll bring this to computer gardening to see if we can set a code style for this for the team.